lib/licensed/sources/nuget.rb

```diff
@@ -184,32 +191,51 @@ def enabled?
       # Ideally we'd use `dotnet list package` instead, but its output isn't
       # easily machine readable and doesn't contain everything we need.
       def enumerate_dependencies
-        json = JSON.parse(project_assets_file)
-        nuget_packages_dir = json["project"]["restore"]["packagesPath"]
-        json["targets"].each_with_object({}) do |(_, target), dependencies|
-          target.each do |reference_key, reference|
-            # Ignore project references
-            next unless reference["type"] == "package"
-            package_id_parts = reference_key.partition("/")
-            name = package_id_parts[0]
-            version = package_id_parts[-1]
-            id = "#{name}-#{version}"
-
-            # Already know this package from another target
-            next if dependencies.key?(id)
-
-            path = File.join(nuget_packages_dir, json["libraries"][reference_key]["path"])
-            dependencies[id] = NuGetDependency.new(
-              name: id,
-              version: version,
-              path: path,
-              metadata: {
-                "type" => NuGet.type,
-                "name" => name
-              }
-            )
-          end
-        end.values
+        reference_keys.map do |reference_key|
+          package_id_parts = reference_key.partition("/")
+          name = package_id_parts[0]
+          version = package_id_parts[-1]
+          id = "#{name}-#{version}"
+
+          path = full_dependency_path(reference_key)
+          error = "Package #{id} not found at any project package folder" if path.nil?
```

Comment:
Won't `path` also be `nil` if for some reason it's missing under libraries -> `reference_key` -> path? Not a big deal but the error will be slightly incorrect.

Reply:
(feel free to ignore, I don't think this is important enough to differentiate on, but perhaps you want to clarify the message)

Reply:
👍  good catch